### PR TITLE
Add led cdevs

### PR DESCRIPTION
**Pull request by sainak**
This adds dkms config and ability to control mute(0x2c), micmute(0x2b) and keyboard_bl(0xf3) from the led subsystem,
I've also added appropriate triggers for mute and micmute but software control is still missing for them

### CODE REVIEW

**Comment by marshevms** [comment]
Hi! You changed code style. Maybe you have a clang-format file?

File: Makefile [hunk 1]
```diff
@@ -1,13 +1,39 @@
...
+else
+KERNEL_BUILD	:= $(KERNEL_MODULES)/build
+endif
+endif
+
+
+all: modules
+
 modules:
-	make -C /lib/modules/$(shell uname -r)/build M=$(shell pwd) modules
```
Comment (BeardOverflow) [inline Makefile:4]: Is this compliant?
https://docs.kernel.org/kbuild/modules.html#command-syntax

`/lib/modules/$(shell uname -r)/build` should point to your kernel's header, no matter what is your operating system.  E.g, for Debian:
```bash
> readlink -f /lib/modules/5.10.0-16-amd64/build
/usr/src/linux-headers-5.10.0-16-amd64
```

File: msi-ec.c [hunk 2]
```diff
@@ -45,7 +48,8 @@ static int ec_read_seq(u8 addr, u8 *buf, int len)
 {
 	int result;
 	u8 i;
-	for (i = 0; i < len; i++) {
+	for (i = 0; i < len; i++)
```
Comment (BeardOverflow) [inline msi-ec.c:51]: Open braces only for functions, not for loop or conditional structures: https://www.kernel.org/doc/html/v5.10/process/coding-style.html#placing-braces-and-spaces

File: msi-ec.c [hunk 3]
```diff
@@ -58,9 +62,9 @@ static int ec_read_seq(u8 addr, u8 *buf, int len)
 // ============================================================ //
 
 static ssize_t charge_control_threshold_show(u8 offset,
-				struct device *device,
-				struct device_attribute *attr,
-				char *buf)
+											 struct device *device,
```
Comment (BeardOverflow) [inline msi-ec.c:65]: I am not sure about if it's correct, but I was checking another kernel modules and the function args only have 8 white-spaces. I am unable to find further information

File: msi-ec.c [hunk 4]
```diff
@@ -724,8 +729,73 @@ static struct platform_driver msi_platform_driver = {
...
 		.name = MSI_DRIVER_NAME,
 	},
-	.probe	= msi_platform_probe,
-	.remove	= msi_platform_remove,
+	.probe = msi_platform_probe,
+	.remove = msi_platform_remove,
+};
+
+// ============================================================ //
+// Sysfs leds subsystem
```
Comment (BeardOverflow) [inline msi-ec.c:737]: Great job! I don't know the existence of this subsystem. Could you add the documentation about how to use it in the README file?

**Review by BeardOverflow — CHANGES_REQUESTED** [review]

File: msi-ec.c (hunk 4 above)
Comment (sainAk) [reply msi-ec.c:744]: sure

File: Makefile (hunk 1 above)
Comment (sainAk) [reply Makefile:4]: yes it picks the headers path based on the distro

Comment (BeardOverflow) [reply Makefile:4]: My mistake. I did a bad quoted. Please see: https://github.com/sainAk/msi-ec/blob/056dff5a8e1cf66efdba74e6a3522ae8849435ac/Makefile#L10-L20

What is the benefit of doing it this way?

```bash
# Ubuntu 22.04
> readlink -f /lib/modules/5.15.0-46-generic/build 
/usr/src/linux-headers-5.15.0-46-generic
```
```
# Fedora 36
> readlink -f /lib/modules/5.18.18-200.fc36.x86_64/build
/usr/src/kernels/5.18.18-200.fc36.x86_64
```

File: Makefile [hunk 5]
```diff
@@ -1,13 +1,39 @@
+VERSION         := 0.08
+TARGET          := $(shell uname -r)
+DKMS_ROOT_PATH  := /usr/src/msi_ec-$(VERSION)
```
Comment (BeardOverflow) [inline Makefile:3]: `/usr/src` is a restricted path for the distro and we shouldn't touch it.
We can write safely in `/usr/local/src`

See Linux Filesystem Hierarchy: https://tldp.org/LDP/Linux-Filesystem-Hierarchy/html/usr.html

Comment (BeardOverflow) [reply Makefile:3]: Sorry, forget it. DKMS works under `/usr/src` path.

**Pull request merged**